newlib/libc/include/stdint.h

```diff
@@ -457,6 +457,17 @@ typedef __uint_least64_t uint_least64_t;
 #endif
 #endif
 
-_END_STD_C
+#if __STDC_WANT_LIB_EXT1__ == 1
+#include <sys/_types.h>
 
-#endif /* _STDINT_H */
+  // could be defined by the user
+#ifndef RSIZE_MAX
+#define RSIZE_MAX SIZE_MAX
+#endif
+
+  typedef __rsize_t rsize_t;
```

Comment:
Looks like this is supposed to be in stddef.h instead of stdint.h? You should be able to remove the include of <sys/_types.h> from this file as that won't be needed.